packages/core/src/ceramic.ts

```diff
@@ -175,9 +196,17 @@ class Ceramic implements CeramicApi {
       await ceramic.setDIDProvider(config.didProvider)
     }
 
+    await ceramic.init()
     return ceramic
   }
 
+  /**
+   * Initialize the Ceramic API
+   */
+  async init(): Promise<void> {
```

Comment:
Real question: what's the benefit of having a separate init() method from the constructor in typescript?  I almost added this myself during an earlier commit I was working on as it is a very common pattern in C++, but in C++ it's bad practice to do i/o within constructors as throwing exceptions from constructors has weird semantics.  So in the C++ world, having a separate initializer method that can do i/o after the constructor has finished is very useful.  But in typescript we seem to do tons of i/o within our constructors all over the place, so it's less clear to me what the advantage of a separate initializer method is?

CC @PaulLeCam in case he has any insight here

Reply:
You cannot `await` in the `constructor`. One of the two common patterns here are `init` and `builder`. I've used `init` in many places but I believe that there are some workarounds as well. See link [how-to-write-an-async-class-constructor-in-typescript-javascript](https://medium.com/better-programming/how-to-write-an-async-class-constructor-in-typescript-javascript-7d7e8325c35e).

Reply:
Ah there it is, that makes sense.  That article you linked is paywalled, but the explanation that init() methods are important for making async function calls makes sense.  Thanks!

Reply:
np! btw, you can bypass the paywall using the **incognito** mode 🙂